pkg/controller/controller.go

```diff
@@ -265,6 +270,39 @@ func (c *Controller) scaleNodeGroup(nodegroup string, nodeGroup *NodeGroupState)
 		return nodeGroup.scaleUpLock.requestedNodes, nil
 	}
 
+	// If we are not locked, we're either init or after a scale event
+	// If the last scale event was a scale out; i.e. nodesDelta > 0
+	// Calculate the k8s registration latencey from cloud provider
+	// resource instantiation
+
+	// Concerned about:
+	// - lock may open before new nodes register and they will be missed
+	if nodeGroup.scaleDelta > 0 {
+		countNewNodes := 0
+
+		for key, nodeInfo := range nodeGroup.NodeInfoMap {
+			nodeRegTime := nodeInfo.Node().ObjectMeta.CreationTimestamp.Time
+			// Check if node registration time newer than last scale out
+			if nodeRegTime.Sub(nodeGroup.lastScaleOut) > 0 {
+				node := nodeInfo.Node()
+				instance, err := c.cloudProvider.GetInstance(node)
+				if err != nil {
+					nodeInstTime := instance.InstantiationTime()
+					nodeRegistrationLag := nodeRegTime.Sub(nodeInstTime)
```

Comment:
This could easily be simplified to `nodeRegTime.Sub(instance.InstantiationTime())`